service/frontend/nexus_handler.go

```diff
@@ -221,6 +237,21 @@ func (c *operationContext) interceptRequest(ctx context.Context, request *matchi
 		return converted
 	}
 
+	// THIS MUST BE THE LAST STEP IN interceptRequest.
+	// Sanitize headers.
+	if request.GetRequest().GetHeader() != nil {
+		sanitizedHeaders := make(map[string]string, len(request.Request.Header))
+		headersBlacklist := c.headersBlacklist.Get()
+		for name, value := range request.Request.Header {
```

Comment:
Wanna use `maps.DeleteFunc` instead, it'll save cloning the map and we are mutating anyways.

Reply:
I didn't want to mutate `request.Request.Header` because it's the same reference as in `options.Header` and the `header` argument in this function. I wasn't sure where else the `options.Header` is used, and wanted to be safe here.

Reply:
Hmm.. I don't think there are any negative implications but let's keep what you have just in case.
You can add a comment so there's some record of why this decision was made.